File: src/main/java/com/example/skyr/InstantFactory.java
package com.example.skyr;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;
import java.time.Instant;
import java.time.ZoneOffset;
import java.time.temporal.ChronoUnit;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class InstantFactory {
    public static Instant create() {
        return Instant.now().truncatedTo(ChronoUnit.MILLIS);
    }

    public static Instant create(final Timestamp timestamp) {
        return timestamp.toLocalDateTime().toInstant(ZoneOffset.UTC);
    }
}

File: src/main/kotlin/com/example/skyr/InstantFactory.kt
package com.example.skyr

import java.time.Instant
import java.time.temporal.ChronoUnit

class InstantFactory private constructor() {

    companion object {
        fun create(): Instant {
            return Instant.now().truncatedTo(ChronoUnit.MILLIS)
        }
    }
}
